src/Umbraco.Core/Scoping/Scope.cs

```diff
@@ -363,19 +361,22 @@ public void Dispose()
             }
 
             // Decrement the lock counters on the parent if any.
-            if (ParentScope != null)
+            ClearLocks(InstanceId);
+            if (ParentScope is null)
             {
-                lock (_dictionaryLocker)
+                // We're the parent scope, make sure that locks of all scopes has been cleared
+                // Since we're only reading we don't have to be in a lock
+                if (ReadLocks?.Count > 0 || WriteLocks?.Count > 0)
                 {
-                    foreach (var readLockPair in ReadLocks)
-                    {
-                        DecrementReadLock(readLockPair.Key, readLockPair.Value);
-                    }
-
-                    foreach (var writeLockPair in WriteLocks)
-                    {
-                        DecrementWriteLock(writeLockPair.Key, writeLockPair.Value);
-                    }
+                    // Dump the dicts into a message for the locks.
+                    StringBuilder builder = new StringBuilder();
```

Comment:
I would move this to a private method like `string GenerateExceptionMessage()`

Reply:
Good idea, I wonder if i should include the entire message in the exception as well? Currently, the string built with the StringBuilder gets logged and then there is a "see log for more details."

Reply:
I think it is okay just to point to the log for more info